test/ATokenFlexVotingFork.t.sol

```diff
@@ -1658,6 +1581,62 @@ contract CastVote is AaveAtokenForkTest {
       // forgefmt: disable-end
     }
   }
+
+  function _testVotesCanBeCastIncrementally(
+    address _userA,
+    address _userB,
+    uint8 _supportTypeA,
+    uint8 _supportTypeB
+  ) private {
+    require(_supportTypeA != _supportTypeB, "This test assumes the support types are different");
+    uint256 _weightA = 1 ether;
+    uint256 _weightB = 3 ether;
+
+    // Deposit some funds.
+    _mintGovAndSupplyToAave(_userA, _weightA);
+    _mintGovAndSupplyToAave(_userB, _weightB);
+
+    // Create the proposal.
+    uint256 _proposalId = _createAndSubmitProposal();
+
+    // UserA expresses a voting preference on the proposal.
+    vm.prank(_userA);
+    aToken.expressVote(_proposalId, _supportTypeA);
+
+    // Submit votes on behalf of the pool.
+    aToken.castVote(_proposalId);
+
+    (uint256 _againstVotes, uint256 _forVotes, uint256 _abstainVotes) =
+      governor.proposalVotes(_proposalId);
+
+    if (_supportTypeA == uint8(VoteType.For)) assertEq(_forVotes, _weightA);
+    if (_supportTypeA != uint8(VoteType.For)) assertEq(_forVotes, 0);
+    if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weightA);
+    if (_supportTypeA != uint8(VoteType.Against)) assertEq(_againstVotes, 0);
+    if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weightA);
+    if (_supportTypeA != uint8(VoteType.Abstain)) assertEq(_abstainVotes, 0);
+
+    // UserA should not be able to express votes again.
+    vm.prank(_userA);
+    vm.expectRevert("already voted");
+    aToken.expressVote(_proposalId, _supportTypeA);
+
+    // UserB expresses a voting preference on the proposal.
+    vm.prank(_userB);
+    aToken.expressVote(_proposalId, _supportTypeB);
+
+    // Submit votes on behalf of the pool.
+    aToken.castVote(_proposalId);
+
+    (_againstVotes, _forVotes, _abstainVotes) = governor.proposalVotes(_proposalId);
+
+    if (_supportTypeA == uint8(VoteType.For)) assertEq(_forVotes, _weightA);
+    if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weightA);
+    if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weightA);
+    if (_supportTypeB == uint8(VoteType.For)) assertEq(_forVotes, _weightB);
+    if (_supportTypeB == uint8(VoteType.Against)) assertEq(_againstVotes, _weightB);
+    if (_supportTypeB == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weightB);
```

Comment:
There will always be an unused support type since there's only two voters here, should we have if statements + assertions to ensure that third type has zero votes?

Reply:
Good point. Clarified the assertions in 3976218